frontend/src/components/userPage/UserPageContent.jsx

```diff
@@ -316,6 +361,66 @@ const UserPageContent = ({ user }) => {
+                                    Cellphone number:
+                                </label>
+                                <input
+                                    type="tel"
+                                    className="border rounded w-full py-2 px-3 text-gray-700 leading-tight"
+                                    value={cellphoneNumber ?? ''}
+                                    placeholder="(00) 00 00000000"
+                                    onChange={(e) => {
+                                        setCellphoneNumber(e.target.value)
+                                    }}
+                                    required
+                                />
+                            </div>
+                            <button
+                                className="bg-blue-700 hover:bg-blue-950 text-white font-bold py-1 px-2 rounded mr-[1rem] mb-[.5rem]"
+                                type="button"
+                                onClick={handlePatchCellphone}
+                            >
+                                Change cellphone number
+                            </button>
+                            <button
+                                className="bg-red-700 hover:bg-blue-950 text-white font-bold py-1 px-2 rounded mr-[1rem] mb-[.5rem]"
+                                type="button"
+                                onClick={handleDeleteCellphone}
+                            >
+                                Delete cellphone number
+                            </button>
+                        </div>
+                    )}
+                    {user.cellphoneNumber == null && (
+                        <div>
+                            <div className="mb-4">
+                                <label className="block text-gray-800 text-sm font-bold mb-2">
+                                    Cellphone number:
+                                </label>
+                                <input
+                                    type="text"
+                                    className="border rounded w-full py-2 px-3 text-gray-700 leading-tight"
+                                    value={cellphoneNumber ?? ''}
+                                    placeholder="(00) 00 00000000"
+                                    onChange={(e) => {
+                                        setCellphoneNumber(e.target.value)
+                                    }}
+                                    required
+                                />
+                            </div>
+                            <button
+                                className="bg-blue-700 hover:bg-blue-950 text-white font-bold py-1 px-2 rounded mr-[1rem] mb-[.5rem]"
+                                type="button"
+                                onClick={handlePatchCellphone}
+                            >
+                                Register for RideSharing
```

Comment:
So there is another button for RideSharing register. I thought ridesharing register is reflected as phone number provided by the user? There are essentially 3 buttons all used to update user's cellphone number. It's very confusing.

Reply:
Yes, I've solved it